### PR TITLE
WIP #2641 - update ProgramDonorAndFileCountsTable to react table 8

### PR DESCRIPTION
**Pull request by samrichca**
# Description of changes

update ProgramDonorAndFileCountsTable to React Table 8. this table is visible at program/TEST-PR/

no uikit PR

## Type of Change

- [ ] Bug
- [x] Dependency updates
- [ ] Feature
- [x] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

**Comment by joneubank** [comment]
Can we leave this open as a draft PR?
We can resolve it later once the new UIKit is published.

I have concerns with the columns being `any[]` type...

**Comment by samrichca** [comment]
sure @joneubank

**Comment by ciaranschutte** [comment]
went down a rabbit hole this weekend on uikit & tables. can you link me to wrap up ticket please/

File: components/pages/program-entity/ProgramCardsLayout/ProgramDonorAndFileCountsTable/index.tsx [hunk 1]
```diff
@@ -17,45 +17,29 @@
...
-  }));
-
-  return (
-    <div
+const ProgramDonorAndFileCountsTable = ({
+  columns,
+  data,
+  title,
+}: {
+  columns: any[]; // can't use generics for column types
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply components/pages/program-entity/ProgramCardsLayout/ProgramDonorAndFileCountsTable/index.tsx:27]: went down a rabbit hole this weekend on uikit & tables. can you link me to wrap up ticket please? @samrichca

**Comment by samrichca** [comment]
@ciaranschutte we don't have a follow-up ticket per se, but here's the epic: https://github.com/icgc-argo/platform-ui/issues/2529

**Pull request closed without merging**